tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.cc

```diff
@@ -458,22 +564,122 @@ Status HorizontalLoopFusionImpl::CreateFusedComputation(
+      bound_operands->push_back(bound_opnd);
+    }
+  }
+  // Always create a dummy tuple instruction to serve as the root of the
+  // computation, as the existence of a root instruction is required by the
+  // HloComputation. The real root instruction will replace it below.
+  HloInstruction* dummy_root = b.AddInstruction(
+      HloInstruction::CreateTuple(std::vector<HloInstruction*>{}));
+  *uniq_computation = b.Build(dummy_root);
+  HloComputation* comp = uniq_computation->get();
+
+  // Preparing clone_map, which maps old operand to new operand.
+  absl::flat_hash_map<const HloInstruction*, HloInstruction*> clone_map;
+  size_t new_param_id = 0;
+  for (size_t i = 0; i < fused_fusion_instrs.size(); ++i) {
+    auto old_params = fused_fusion_instrs[i]->fused_parameters();
+    for (size_t j = 0; j < old_params.size(); ++j) {
+      HloInstruction* old_param = old_params[j];
+      HloInstruction* new_param = comp->parameter_instruction(new_param_id++);
+      clone_map.insert({old_param, new_param});
+    }
+  }
+
+  // Clone every fused computation.
+  for (size_t i = 0; i < fused_fusion_instrs.size(); ++i) {
+    auto def_to_use_order = fused_fusion_instrs[i]
+                                ->fused_instructions_computation()
+                                ->MakeInstructionPostOrder();
+    for (HloInstruction* old_instr : def_to_use_order) {
+      if (old_instr->opcode() == HloOpcode::kParameter) {
+        // Parameters have been created.
+        continue;
+      }
+      std::vector<HloInstruction*> new_opnds;
+      const auto& old_opnds = old_instr->operands();
+      new_opnds.reserve(old_opnds.size());
+      for (HloInstruction* old_opnd : old_opnds) {
+        CHECK(clone_map.find(old_opnd) != clone_map.end());
+        new_opnds.push_back(clone_map[old_opnd]);
+      }
+      HloInstruction* new_instr = comp->AddInstruction(
+          old_instr->CloneWithNewOperands(old_instr->shape(), new_opnds));
+      clone_map.insert({old_instr, new_instr});
+    }
+  }
+
+  // Since we require each fusion to have the same number of outputs, we can
+  // simply use the first fusion as the representative for output size.
+  size_t fused_instr_output_size =
+      GetOutputSizeOfFusible(*fused_fusion_instrs[0]);
+  std::vector<HloInstruction*> tuple_operands(fused_instr_output_size * 
+                                              fused_fusion_instrs.size());
+  for (size_t i = 0; i < fused_instr_output_size; ++i) {
+    for (size_t j = 0; j < fused_fusion_instrs.size(); ++j) {
+      const HloInstruction* old_output =
+          GetOutputsOfFusible(*fused_fusion_instrs[j])[i];
+      HloInstruction* new_output = clone_map[old_output];
+      tuple_operands[fused_instr_output_size * j + i] = new_output;
+    }
+  }
+
+  // Make a tuple instruction of fused instruction outputs as 
+  // the root of fused computation.
+  HloInstruction* tuple =
+      comp->AddInstruction(HloInstruction::CreateTuple(tuple_operands));
```

Comment:
The added instruction is missing the metadata (we could use any from the fused instructions), please see how the sibling method copies it.

Reply:
Fixed